pkg/telemetryquerylanguage/contexts/metrics/metrics.go

```diff
@@ -807,15 +807,15 @@ func accessPositiveBucketCounts() pathGetSetter {
 		getter: func(ctx tql.TransformContext) interface{} {
 			switch ctx.GetItem().(type) {
 			case pmetric.ExponentialHistogramDataPoint:
-				return ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().MBucketCounts()
+				return ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().BucketCounts().AsRaw()
 			}
 			return nil
 		},
 		setter: func(ctx tql.TransformContext, val interface{}) {
 			if newPositiveBucketCounts, ok := val.([]uint64); ok {
 				switch ctx.GetItem().(type) {
 				case pmetric.ExponentialHistogramDataPoint:
-					ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().SetMBucketCounts(newPositiveBucketCounts)
+					ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().SetBucketCounts(pcommon.NewImmutableUInt64Slice(newPositiveBucketCounts))
```

Comment:
@TylerHelmuth these are unnecessary extra copies during the call to "NewImmutable", I know you did some smartness for TraceID, we should do the same for this, in a separate PR.

Reply:
Good idea, can you open an issue with the details?

Reply:
I was trying to delegate that to you :))

Reply:
![image](https://user-images.githubusercontent.com/12352919/182227274-169dbe50-3e19-4efe-b9ef-9930d4d97add.png)

Reply:
I'll take it since I own the package.

Reply:
done: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12872